Adding xapi js tracker

jujusb opened the pull request.

jujusb committed:
adding server on localhost to localdev

jujusb committed:
adding xAPI for SG Tracker

jujusb committed:
adding first statement start game

jujusb committed:
fix tracker online + uri

jujusb committed:
fix tracker

jujusb committed:
adding traces tracking

jujusb committed:
in debug

fix game day and hour + notification amount

jujusb committed:
fix some localization in french

jujusb committed:
adding extension game day and game hour and mobile messages for more info

jujusb committed:
fix set result when empty

fix xapi tracker

jujusb committed:
adding new statements to track

jujusb committed:
fix some statements

jujusb committed:
fix some french localization

jujusb committed:
offline mode

jujusb committed:
adding .github/workflows/jekyll-gh-pages.yml

jujusb committed:
change url for game

jujusb committed:
remove test statement

jujusb committed:
adding debug option

jujusb committed:
fix debug

jujusb committed:
fix debug

jujusb committed:
adding progress for scene

jujusb committed:
fix auth token + adding duration in second in completion

jujusb committed:
fix initialized game

jujusb committed:
adding duration to result statement

jujusb committed:
adding completed statement for test

jujusb committed:
fix default uri

jujusb committed:
fix startedtime for test

jujusb committed:
fix default uri for result statements

jujusb committed:
fix result statements

jujusb committed:
fix completedGame() function send

jujusb committed:
fix repeated day

jujusb committed:
fix send statements

jujusb committed:
fix some localization in french

jujusb committed:
fix send statements

jujusb committed:
Adding progress through days for game completion

jujusb committed:
Update gameManager.js

jujusb committed:
Update bedroomBase.js

jujusb committed:
Update friendsTab.js

jujusb committed:
adding rollup config

jujusb committed:
fix rollup

jujusb committed:
Delete server.js

jujusb committed:
fix send completed trace

jujusb committed:
adding nightmare traces

jujusb committed:
fix localization in french

jujusb committed:
fix completion

jujusb committed:
adding jstracker as bundle

jujusb committed:
fix bundle

jujusb committed:
adding bundle instead of tracker folder

jujusb committed:
fix bundle

jujusb committed:
fix progress completed game

jujusb committed:
fix exit send

jujusb committed:
fix send backup statements

MattCastUCM committed:
click spam handling, visual fixes

gorco committed:
Merge pull request #2 from MattCastUCM/adding-xapi-js-tracker

some fixes: click spam handling and visual improvements

jujusb committed:
fix bundle

fix localization

fix completed with sendBatch + adding params for batch

update using enqueue instead of sendStatement to have more send

fix oauth2 parameters needed

fix xapi js from simva received data

fix using function generateXAPITrackerFromURLParams instead

fix default uri in bundle

fix bundle

fix bundle xapitracker

fix bundle

fix bundle

removing exit menu in language menu + fix bundle

fix bundle

jujusb committed:
Merge branch 'main' into adding-xapi-js-tracker

jujusb committed:
fix localization in french + adding new line to be able to read correctlyu

imartinezortiz commented on package.json:
```diff
@@ -0,0 +1,38 @@
+    "url": "git+https://github.com/e-ucm/ConectadoWeb.git"
+  },
+  "author": "e-UCM Team",
+  "license": "GNU GENERAL PUBLIC LICENSE",
```
Apache-2.0

imartinezortiz commented on logs.js:
Check if this is really used anywhere in the code or package.json as a cli script. If it is not used remove. If it is used, figure it out how to remove it :D

jujusb committed:
send batch directly when initialized and progressed

jujusb committed:
fix initialized before completed and await in language menu

jujusb committed:
fix conectadoWeb tracker new version

jujusb committed:
fix all with using statement builder instead of enqueue

MattCastUCM committed:
Menús, interfaces y día 1 en chino

pedrolemi committed:
Prueba Readme

MattCastUCM committed:
Días 2 y 5 en chino

pedrolemi committed:
Traducción archivos generales, día 1 y días 2 comenzado

Boa noite galera 😪

MattCastUCM committed:
Traducción en chino terminada y correcciones

MattCastUCM committed:
Correcciones, textos generales en cantonés

MattCastUCM committed:
Traducción en cantonés completa

MattCastUCM committed:
Usar versión en portugués actualizada

MattCastUCM committed:
Versión en portugués terminada

jujusb committed:
tracker-v2

MattCastUCM committed:
Revisión de textos

jujusb committed:
fix dialog manager for have the same dialog at launch than at completed dialog

jujusb committed:
fix duration

MattCastUCM committed:
Traducciones funcionando en el juego

pedrolemi committed:
Arreglar error

pedrolemi committed:
Revert "Arreglar error"

This reverts commit e223f4f9713b40a6c2575e6172481e0ed13da4ae.

pedrolemi committed:
Arreglos género portugués

jujusb committed:
adding send when finishing

jujusb committed:
fix with last version of tracker

jujusb committed:
fix start and login

MattCastUCM committed:
Corrección

jujusb committed:
fix with tracker v1.1.0

jujusb committed:
Merge branch 'tracker-v2' into pt-cn

jujusb committed:
Merge pull request #6 from MattCastUCM/pt-cn

Adding Brazilian Portuguese, Chinese and Cantonese translations

jujusb committed:
fix use last js-tracker version

jujusb committed:
Merge pull request #4 from e-ucm/tracker-v2

Tracker v2

jujusb committed:
Merge pull request #3 from e-ucm/statement-builder

fix all with using statement builder instead of enqueue

jujusb committed:
fix readme

tyomikjan13 committed:
Language selection successfully sent to OGD

gorco committed:
Fixed a small error in a variable that was not changed in the last refactoring of the tracker. Some things were not tested and should have been, right?.

tyomikjan13 committed:
override send statement for OGD

tyomikjan13 committed:
xapi login compilation without errors

tyomikjan13 committed:
param to turn Open Game Data integration on/off

tyomikjan13 committed:
?wisconsin=true sends data to OGD

gorco committed:
Fixed a bug that prevented playing on day 4 in English.

gorco committed:
minifix

tyomikjan13 committed:
ogd - avoid double initialization. actor id generation. utf-8 encoding fixed.

tyomikjan13 committed:
shorten event_name to fit 32 chars

tyomikjan13 committed:
Merge branch 'e-ucm:adding-xapi-js-tracker' into adding-xapi-js-tracker

tyomikjan13 committed:
ogd - event_name = object_type + verbId. unicode escaping.

jujusb committed:
Limit dialog text length for xAPI tracking to 25 characters

jujusb committed:
Add xAPI dialog tracking functionality to DialogManager

tyomikjan13 committed:
ogd - tracking game state

tyomikjan13 committed:
Merge branch 'e-ucm:adding-xapi-js-tracker' into adding-xapi-js-tracker

tyomikjan13 committed:
fix - first event sent in bootScene.start

jujusb committed:
Refactor various components for improved xAPI tracking and consistency

jujusb committed:
Add 'update-module' branch to workflow triggers

tyomikjan13 committed:
Merge branch 'e-ucm:adding-xapi-js-tracker' into adding-xapi-js-tracker

tyomikjan13 committed:
ogd - avoid game state keys duplicating in event data

tyomikjan13 committed:
ogd - fix comment

jujusb committed:
Merge pull request #7 from tyomikjan13/adding-xapi-js-tracker

Adding Open Game Data tracker

jujusb committed:
adding dist bundle

MattCastUCM committed:
Usar minify de phaser y cambiar plugin de i18next por versión local

jujusb committed:
Remove logs.js and update license information in package.json and package-lock.json to Apache-2.0

jujusb committed:
fix: update script paths and improve circular dependency handling

- Changed script paths in index.html and dist/index.html to remove leading './'.
- Updated phaser.js to phaser.min.js for optimized loading.
- Added onwarn function in rollup configuration files to handle circular dependency warnings specifically for 'node_modules/opengamedata-js-log/'.
- Refactored clearRunningScenes method in gameManager.js to simplify shutdown logic for scenes.
